lib/libzfs/os/linux/libzfs_mount_os.c

```diff
@@ -336,6 +611,9 @@ do_mount(const char *src, const char *mntpt, char *opts, int flags)
 int
 do_unmount(const char *mntpt, int flags)
 {
+	if (! libzfs_envvar_is_set("ZFS_MOUNT_HELPER")) {
```

Comment:
nit: no space after `!`, and I agree no need for an `else` here.  It's simpler without, though please add a newline after the `if` conditional's closing `}`.